ipn/ipnlocal/local.go

```diff
@@ -1300,6 +1301,29 @@ func (b *LocalBackend) updateNetmapDeltaLocked(muts []netmap.NodeMutation) (hand
 // setExitNodeID updates prefs to reference an exit node by ID, rather
 // than by IP. It returns whether prefs was mutated.
 func setExitNodeID(prefs *ipn.Prefs, nm *netmap.NetworkMap) (prefsChanged bool) {
+	if exitNodeIDStr, _ := syspolicy.GetString(syspolicy.ExitNodeID, ""); exitNodeIDStr != "" {
+		exitNodeID := tailcfg.StableNodeID(exitNodeIDStr)
+		if prefs.ExitNodeID != exitNodeID {
+			prefs.ExitNodeID = tailcfg.StableNodeID(exitNodeID)
+			prefs.ExitNodeIP = netip.Addr{}
+			return true
+		}
+	}
+
+	var oldExitNodeID tailcfg.StableNodeID
```

Comment:
While I think it's going to be harmless in the end to only set `oldExitNodeID` in the "syspolicy has an `ExitNodeIP` configured" path, it took a bunch of thinking it through to confirm that. I think it would be simpler just to always capture the old exit node ID.

```suggestion
	oldExitNodeID := prefs.ExitNodeID
```
and then don't bother setting it on line 1320